client/block-builder/src/lib.rs

```diff
@@ -237,7 +232,7 @@ where
 				ExecutionContext::BlockConstruction,
 				inherent_data
 			))
-		})
+		}).map_err(|e| Error::Application(Box::new(e)))
```

Comment:
I would prefer adding another error variant, so `::Application` can cease to exist eventually.